client/src/components/related/Outfit.jsx

```diff
@@ -2,57 +2,95 @@ import React from 'react';
+      let newOutfit = [ selectedProduct.id, ...outfit ];
+      localStorage.setItem('outfit', JSON.stringify(newOutfit));
+      console.log('Outfit saved to localStorage:', localStorage.getItem('outfit'));
+
+      this.setState({ outfit: newOutfit });
     }
-    const newOutfit = [ selectedProduct, ...outfit ];
-    localStorage.setItem('outfit', JSON.stringify(newOutfit.map(item => item.id)));
-    console.log('Outfit saved to localStorage:', localStorage.getItem('outfit'));
-
-    updateOutfit(newOutfit);
-  };
-
-  const removeFromOutfit = (product) => {
-    const match = product.id;
-    const newOutfit = [ ...outfit ];
-    console.log('Remove', match);
-    let i = newOutfit.length;
-    while (i--) {
-      if (newOutfit[i].id === match) {
-        newOutfit.splice(i, 1);
-      }
+  }
+
+  removeFromOutfit(productId) {
+    const { localStorage } = window;
+    const { outfit } = this.state;
+    let newOutfit = [ ...outfit ];
+    const removalIndex = newOutfit.indexOf(productId);
+
+    if (removalIndex > -1) {
+      console.log(`Remove ${productId} from outfit`);
+      newOutfit.splice(removalIndex, 1);
+
+      newOutfit.length ? localStorage.setItem('outfit', JSON.stringify(newOutfit)) : localStorage.removeItem('outfit');
+      this.setState({ outfit: newOutfit });
     }
+  }
+
+  componentDidMount() {
+    this.loadOutfit();
+  }
+
+  render() {
+    const { addToOutfit, removeFromOutfit } = this;
+    const { outfit } = this.state;
+    const { selectedProduct, selectProduct, checkCache, updateCache } = this.props;
 
-    newOutfit.length ? localStorage.setItem('outfit', JSON.stringify(newOutfit.map(item => item.id))) : localStorage.removeItem('outfit');
-    updateOutfit(newOutfit);
-  };
-
-  let key = 0;
-  return selectedProduct ? (
-    <div id='Outfit'>
-      <h1></h1>
-      <span className='rp-component-title'>YOUR OUTFIT</span>
-      <div className='rp-card-container'>
-        <div className='rp-card rp-card-placeholder' title={`Add ${selectedProduct.name} to outfit`} onClick={ addToOutfit }>
-          <h1>+</h1>
-          <h2>Add to Outfit</h2>
+    let key = 0;
+    return selectedProduct ? (
+      <div id='Outfit'>
+        <h1></h1>
+        <span className='rp-component-title'>YOUR OUTFIT</span>
+        <div className='rp-card-container'>
+          <div className='rp-card rp-card-placeholder add-to-outfit' title={`Add ${selectedProduct.name} to outfit`} onClick={ addToOutfit }>
+            <h1>+</h1>
+            <h2>Add to Outfit</h2>
+          </div>
+          {outfit.length ? (
+            outfit.map(id => {
+              return <ProductCard key={ `outfitCard${id}` }
+                type='Outfit'
+                // value={ product.id }
+                productId={ id }
+                selectProduct={ selectProduct }
+                action={ removeFromOutfit }
+                checkCache={ checkCache }
+                updateCache={ updateCache }
+              />;
+            })) : null
```

Comment:
Curious here. In some earlier code I saw you using && rather than ternaries. I have mostly used && for conditions of only rendering a component if a case is met unless an alternative was to be rendered instead. Is there a reason for the switch? Or am I missing something here on using one method vs. another for triggering rendering pieces?

Reply:
I thought React expected `null` rather than `undefined`. Happy to be wrong there.